apis/core/v1alpha1/resource.go

```diff
@@ -197,7 +197,7 @@ type ResourceSpec struct {
 
 	// ProviderReference specifies the provider that will be used to create,
 	// observe, update, and delete this managed resource.
-	ProviderReference *corev1.ObjectReference `json:"providerRef"`
+	ProviderReference *Reference `json:"providerRef"`
```

Comment:
Shouldn't this field be required actually?

```suggestion
	ProviderReference Reference `json:"providerRef"`
```

If it wasn't a pointer type, then we'd not risk having [nil pointer dereference](https://github.com/crossplane/provider-aws/pull/261) when we use it, like here in your other PR https://github.com/crossplane/provider-aws/blob/87a255e5ee6d2b2d58711f20e5d8e199101c39e2/pkg/controller/cache/cachesubnetgroup/controller.go#L86

Reply:
Great point! I'll update here :)